src/workflow_engine/core/data.py

```diff
@@ -8,15 +8,13 @@ class Data(BaseModel):
     model_config = ConfigDict(extra="forbid", frozen=True)
 
 
-Input = TypeVar("Input", bound=Data)
 Input_contra = TypeVar("Input_contra", bound=Data, contravariant=True)
 
-Output = TypeVar("Output", bound=Data)
 Output_co = TypeVar("Output_co", bound=Data, covariant=True)
 
 
 __all__ = [
     "Data",
-    "Input",
-    "Output",
+    "Input_contra",
```

Comment:
contra and co?

Is something is missing here

Reply:
just cleaning up some junk in the code, we typically always want to use `contra` and `co` suffixes on Python type variables for generic classes.

`covariant` means that `S is a T` implies `Foo[S] is a Foo[T]`

`contravariant` means that `S is a T` implies `Foo[T] is a Foo[S]`

Covariant is how we usually think about our container types: a `list[PDFFile]` is a `list[File]`

Contravariant is how we usually think about argument types of functions: a `Predicate[File]` is a `Predicate[PDFFile]` because any PDFfile you can pass into the latter can also be passed into the former; so the former serves at least the same functions

Reply:
To finish the mini infodump, Python actually gives special treatment to type variables that end in `_contra` and `_co`, which is why these guys are named weirdly -- it's extra reinforcement on top of the `covariant=True` and `contravariant=True` flags on those types